static/app/views/profiling/profileDetails.tsx

```diff
@@ -149,6 +148,127 @@ function ProfileDetails() {
     [location, search]
   );
 
+  const [filters, setFilters] = useState<Partial<Record<TableColumnKey, string[]>>>({});
+
+  const columnFilters = useMemo(() => {
+    function makeOnFilterChange(key: string) {
+      return values => {
+        setFilters(prevFilters => ({
+          ...prevFilters,
+          [key]: values.length > 0 ? values.map(val => val.value) : undefined,
+        }));
+      };
+    }
+    return {
+      type: {
+        values: ['application', 'system'],
+        onChange: makeOnFilterChange('type'),
+      },
+      image: {
+        values: pluckUniqueValues(slowestFunctions, 'image').sort((a, b) =>
+          a.localeCompare(b)
+        ),
+        onChange: makeOnFilterChange('image'),
+      },
+    };
+  }, [slowestFunctions]);
+
+  const currentSort = useMemo<GridColumnSortBy<TableColumnKey>>(() => {
+    let key = location.query?.functionsSort ?? '';
+
+    const defaultSort = {
+      key: 'self weight',
+      order: 'desc',
+    } as GridColumnSortBy<TableColumnKey>;
+
+    const isDesc = key[0] === '-';
+    if (isDesc) {
+      key = key.slice(1);
+    }
+
+    if (!key || !tableColumnKey.includes(key as TableColumnKey)) {
+      return defaultSort;
+    }
+
+    return {
+      key,
+      order: isDesc ? 'desc' : 'asc',
+    } as GridColumnSortBy<TableColumnKey>;
+  }, [location.query]);
+
+  useEffect(() => {
+    const removeListener = browserHistory.listenBefore((nextLocation, next) => {
```

Comment:
Why is this effect required?

Reply:
this is addressing something we've encountered before / is a larger problem; sticky querystrings 

i based some off this off existing code we had for sorting. we track the `functionsSort` key on the querystring to determine what needs to be sorted. one of the columns is `self weight` however going back to a previous route that also looks for this querystring would result in it just passing it's value off the the backend and making the server throw an error.

of course, there are a large number of ways to solve this. simplest being just use a different key... and yeah that works, but id argue its a crap solution since it doesn't solve the root of the issue.

i was originally going to convert this into a hook to just handle querystrings and the cleanup between routes using this `useEffect` but i didn't want to get too cute on my PR and takeaway from what i'm trying to solve for.

its something i'd rather cut a PR for and socialize in TSC.